### PR TITLE
Sector X Overlay Decompiled

### PR DESCRIPTION
**Pull request by KiritoDv**
image

DESTROY
DESTROY
I MUST BE COMPLETE

Co-authored-by:
@sonicdcer 
@inspectredc 
@petrie911

### CODE REVIEW

File: src/overlays/ovl_i2/fox_sx.c [hunk 1]
```diff
@@ -1,81 +1,1980 @@
...
+            sp34 = fabsf(actor->obj.pos.z - gBosses[0].obj.pos.z) * 0.1f;
+            Math_SmoothStepToF(&actor->obj.pos.x, gBosses[0].obj.pos.x + __sinf(actor->unk_0F4.z * 0.017453292f) * sp34,
+                               0.1f, 20.0f, 0.0f);
+            Math_SmoothStepToF(&actor->obj.pos.y, gBosses[0].obj.pos.y, 0.1f, 20.0f, 0.0f);
+            Math_SmoothStepToF(&actor->obj.pos.z, gBosses[0].obj.pos.z, 1.0f, 55.0f, 0);
+            if (gBosses[0].actionState == 20) {
+                actor->unk_0B6 = 10;
+                break;
+            }
+            if ((!(gFrameCount & 3)) && (Rand_ZeroOne() < 0.2f)) {
```
Comment (sonicdcer) [inline src/overlays/ovl_i2/fox_sx.c:142]:
```suggestion
if (!(gFrameCount & 3)

File: src/overlays/ovl_i2/fox_sx.c [hunk 2]
```diff
@@ -1,81 +1,1980 @@
...
+            }
+            if ((gBosses[0].unk_04C == 0x4B) && (gBosses[0].actionState == 0x55)) {
+                actor->unk_0B6 = 2;
+                actor->vel.x = 10.0f;
+                actor->vel.y = -10.0f;
+                actor->vel.z = 30.0f;
+                func_800BA808(gMsg_ID_4111, RCID_SLIPPY);
+                func_8007A6F0(&actor->obj.pos, 0x2903A008);
+                gBosses[0].swork[6] = 0;
+                *(gTeamShields + 2) = -2;
```
Comment (sonicdcer) [inline src/overlays/ovl_i2/fox_sx.c:153]:
```suggestion
gTeamShields[2] = -2;

**Review by sonicdcer — CHANGES_REQUESTED** [review]

File: src/overlays/ovl_i2/fox_sx.c [hunk 3]
```diff
@@ -1,81 +1,1986 @@
...
+
+        case -1:
+            actor->obj.rot.y = (-gPlayer[gPlayerNum].unk_058 * 180.0f) / M_PI;
+            actor->obj.rot.x = (gPlayer[gPlayerNum].unk_05C * 180.0f) / M_PI;
+            RCP_SetupDL(&gMasterDisp, 0x40);
+            gDPSetPrimColor(gMasterDisp++, 0, 0, 220, 70, 30, 255);
+            Matrix_Scale(gGfxMatrix, 70.0f, 70.0f, 1.0f, 1);
+            Matrix_SetGfxMtx(&gMasterDisp);
+            gSPDisplayList(gMasterDisp++, D_60010C0);
+        default:
```
Comment (sonicdcer) [inline src/overlays/ovl_i2/fox_sx.c:225]: space

File: src/overlays/ovl_i2/fox_sx.c [hunk 4]
```diff
@@ -1,81 +1,1986 @@
...
+    *arg1 = NULL;
+    return 0;
+}
+
+void func_i2_8018FF84(Boss* boss) {
+    Animation_DrawSkeleton(3, &D_6020C68, boss->vwork, func_i2_8018FF40, func_i2_80193208, boss, gCalcMatrix);
+}
+
+s32 func_i2_8018FFDC(s32 arg0, Gfx** arg1, Vec3f* arg2, Vec3f* arg3, void* arg4) {
+    if (((arg0 >= 2) && (arg0 < 0xC)) || ((arg0 >= 0x20) && (arg0 < 0x24))) {
```
Comment (sonicdcer) [inline src/overlays/ovl_i2/fox_sx.c:348]: Decimal

File: src/overlays/ovl_i2/fox_sx.c [hunk 5]
```diff
@@ -1,81 +1,1986 @@
...
+                    }
+                    break;
+                case 0xE5:
+                    if (gTeamShields[3] > 0) {
+                        func_i2_801938D8(&gActors[2], 2);
+                    }
+                    break;
+            }
+
+            if (D_80177A80 >= 0x12D) {
```
Comment (sonicdcer) [inline src/overlays/ovl_i2/fox_sx.c:1603]: dec

**Review by sonicdcer — APPROVED** [review]

File: src/overlays/ovl_i2/fox_sx.c [hunk 6]
```diff
@@ -1,81 +1,1986 @@
...
+                if ((gBossFrameCount < 2200) || (gTeamShields[2] <= 0)) {
+                    if (boss->health > 150) {
+                        boss->actionState = 8;
+                    } else {
+                        boss->actionState = 35;
+                        boss->timer_050 = 400;
+                        boss->unk_078.y = 0.0f;
+                        boss->fwork[44] = 0.0f;
+                        boss->fwork[43] = 0.0f;
+                        boss->fwork[42] = 0.0f;
```
Comment (petrie911) [inline src/overlays/ovl_i2/fox_sx.c:1018]: decreasing order suggests chain assign. Does that work?

File: src/overlays/ovl_i2/fox_sx.c [hunk 7]
```diff
@@ -1,81 +1,1986 @@
...
+            Math_SmoothStepToF(&boss->obj.rot.y, 2880.0f, 0.1f, boss->unk_078.y, 0.1f);
+
+            if ((boss->timer_050 == 0) && (boss->obj.rot.y == 2880.0f)) {
+                boss->obj.rot.y = 0.0f;
+                boss->unk_04C = 0;
+                boss->fwork[0] = 0.0f;
+                if ((gBossFrameCount < 2200) || (gTeamShields[2] <= 0)) {
+                    boss->actionState = 9;
+                } else {
+                    boss->actionState = 0x55;
```
Comment (petrie911) [inline src/overlays/ovl_i2/fox_sx.c:1075]:
```suggestion
                    boss->actionState = 85;
```

File: src/overlays/ovl_i2/fox_sx.c [hunk 8]
```diff
@@ -1,81 +1,1986 @@
...
+    }
+
+    D_i2_80195640 = 1;
+
+    Animation_DrawSkeleton(1, &D_6020C68, boss->vwork, func_i2_80192AF0, func_i2_80193208, boss, &gIdentityMatrix);
+
+    if (((boss->swork[1] != 0) && (boss->swork[3] > 0)) && ((boss->fwork[4] < 45.0f) || (boss->fwork[4] > 315.0f))) {
+        boss->swork[1]++;
+        if (boss->swork[1] >= 41) {
+            if (boss->swork[1] >= 51) {
```
Comment (petrie911) [inline src/overlays/ovl_i2/fox_sx.c:1152]:
```suggestion
        if (boss->swork[1] > 40) {
            if (boss->swork[1] > 50) {
```

File: src/overlays/ovl_i2/fox_sx.c [hunk 9]
```diff
@@ -1,81 +1,1986 @@
...
+            boss->info.hitbox[66] = 70.0f;
+        }
+    } else {
+        boss->info.hitbox = SEGMENTED_TO_VIRTUAL(D_6032768);
+    }
+
+    if (boss->swork[0] == 1) {
+        Math_SmoothStepToF(&boss->fwork[45], 1.0f, 1.0f, 0.05f, 0);
+        return;
+    }
```
Comment (petrie911) [inline src/overlays/ovl_i2/fox_sx.c:1201]: I think this should be else

File: src/overlays/ovl_i2/fox_sx.c [hunk 10]
```diff
@@ -1,81 +1,1986 @@
...
+                arg3->x += boss->fwork[17] * 0.1f;
+                arg3->z += boss->fwork[17] * 0.2f;
+            }
+            break;
+    }
+
+    return 0;
+}
+
+void func_i2_80193208(s32 arg0, Vec3f* arg1, void* ptr) {
```
Comment (petrie911) [inline src/overlays/ovl_i2/fox_sx.c:1324]: Signature of a post limb draw should be `void func_i2_80193208(s32 limbIndex, Vec3f* rot, void* data)`, where data can be renamed to something more fitting.

File: src/overlays/ovl_i2/fox_sx.c [hunk 11]
```diff
@@ -1,81 +1,1986 @@
...
+                y = gActors[5].obj.pos.y - (player->camEye.y + 50.0f);
+                z = gActors[5].obj.pos.z - (player->camEye.z + 20.0f);
+                xzDeg = Math_RadToDeg(Math_Atan2F(x, z));
+                xyzDeg = Math_RadToDeg(-Math_Atan2F(y, sqrtf(SQ(x) + SQ(z))));
+                Matrix_RotateY(gCalcMatrix, M_DTOR * xzDeg, 0);
+                Matrix_RotateX(gCalcMatrix, M_DTOR * xyzDeg, 1);
+                src.y = src.x = 0.0f;
+                src.z = 100.0f;
+                Matrix_MultVec3fNoTranslate(gCalcMatrix, &src, &dest);
+                func_8006EEFC(0x64, 0.0f, player->camEye.y + 50.0f, player->camEye.z + 20.0f, dest.x, dest.y, dest.z,
```
Comment (petrie911) [inline src/overlays/ovl_i2/fox_sx.c:1535]:
```suggestion
                func_8006EEFC(100, 0.0f, player->camEye.y + 50.0f, player->camEye.z + 20.0f, dest.x, dest.y, dest.z,
```

File: src/overlays/ovl_i2/fox_sx.c [hunk 12]
```diff
@@ -1,81 +1,1986 @@
...
+                    func_80079618(gActors[5].obj.pos.x, gActors[5].obj.pos.y, gActors[5].obj.pos.z, 0.5f);
+                }
+                func_8007A6F0(&gActors[5].obj.pos, 0x2903B009);
+                player->unk_1D0 += 1;
+                D_80177A48[0] = 0.1f;
+            }
+            break;
+
+        case 2:
+            if (D_80177A80 == 0xAA) {
```
Comment (petrie911) [inline src/overlays/ovl_i2/fox_sx.c:1555]:
```suggestion
            if (D_80177A80 == 170) {
```
`D_80177A80` is also a frame counter, so should be decimal.

File: src/overlays/ovl_i2/fox_sx.c [hunk 13]
```diff
@@ -1,81 +1,1986 @@
...
+    actor->obj.id = OBJ_ACTOR_195;
+    Object_SetInfo(&actor->info, actor->obj.id);
+    if (arg1 == 3) {
+        actor->unk_0B6 = 1;
+        actor->unk_0B8 = 0x14;
+        actor->obj.rot.x = -player->unk_0E4 - 10.0f;
+        actor->obj.rot.y = (player->unk_0E8 + 180.0f) - 10.0f;
+        actor->fwork[9] = 10.0f;
+        return;
+    }
```
Comment (petrie911) [inline src/overlays/ovl_i2/fox_sx.c:1710]: else?

File: src/overlays/ovl_i2/fox_sx.c [hunk 14]
```diff
@@ -1,81 +1,1986 @@
...
+    actor->obj.rot.z = D_i2_80195770[arg1];
+    actor->iwork[0xB] = 1;
+    actor->iwork[0xE] = D_i2_8019577C[arg1];
```
Comment (petrie911) [inline src/overlays/ovl_i2/fox_sx.c:1713]:
```suggestion
    actor->iwork[11] = 1;
    actor->iwork[14] = D_i2_8019577C[arg1];
```
work indices also in decimal

File: src/overlays/ovl_i2/fox_sx.c [hunk 15]
```diff
@@ -1,81 +1,1986 @@
...
+
+            if (D_80177A80 >= 1196) {
+                D_80178348 = D_80178350 = D_80178354 = 0;
+                D_80178358 = 255;
+                if (D_80178340 == 255) {
+                    player->state_1C8 = PLAYERSTATE_1C8_6;
+                    D_8017837C = 4;
+                    player->timer_1F8 = 0;
+                    func_800A6148();
+                    D_800D3180[2] = Play_CheckMedalStatus(0x96U) + 1;
```
Comment (petrie911) [inline src/overlays/ovl_i2/fox_sx.c:1800]:
```suggestion
                    D_800D3180[2] = Play_CheckMedalStatus(150) + 1;
```

File: src/overlays/ovl_i2/fox_sx.c [hunk 16]
```diff
@@ -1,81 +1,1986 @@
...
+            break;
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i2/fox_sx/func_i2_8018F884.s")
+        case 1090:
+            gActors[2].unk_0B8 = 0xA;
+            break;
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i2/fox_sx/func_i2_8018FA04.s")
+        case 1120:
+            gActors[3].unk_0B8 = 0xA;
```
Comment (petrie911) [inline src/overlays/ovl_i2/fox_sx.c:1856]:
```suggestion
            gActors[3].unk_0B8 = 10;
```

File: src/overlays/ovl_i2/fox_sx.c [hunk 17]
```diff
@@ -1,81 +1,1986 @@
...
+            break;
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i2/fox_sx/func_i2_8018FBBC.s")
+        case 1160:
+            func_800182F4(0x103200FF);
+            func_800182F4(0x113200FF);
+            Audio_PlaySfx(0x09000002U, &player->unk_460, 0U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+            player->unk_1D0 = 2;
+            player->timer_1F8 = 0x28;
```
Comment (petrie911) [inline src/overlays/ovl_i2/fox_sx.c:1864]:
```suggestion
            player->timer_1F8 = 40;
```

File: src/overlays/ovl_i2/fox_sx.c [hunk 18]
```diff
@@ -1,81 +1,1986 @@
...
+    if (boss->swork[0] == 1) {
+        Math_SmoothStepToF(&boss->fwork[45], 1.0f, 1.0f, 0.05f, 0);
+        return;
+    }
+
+    Math_SmoothStepToF(&boss->fwork[45], 0.0f, 1.0f, 0.05f, 0);
+}
+
+s32 func_i2_80192AF0(s32 arg0, Gfx** arg1, Vec3f* arg2, Vec3f* arg3, void* arg4) {
+    Vec3f sp64 = D_i2_80195674;
```
Comment (petrie911) [inline src/overlays/ovl_i2/fox_sx.c:1207]:
```suggestion
    Vec3f sp64 = {0.0f, 0.0f, 0.0f};
```
This sort of thing is basically always a stack variable intializer. You'll have to move the data around to accommodate this.

**Review by petrie911 — APPROVED** [review]

**Pull request merged**